.github/workflows/build-wheel-macos-x86_64.yaml

```diff
@@ -54,23 +52,23 @@ jobs:
       uses: actions/cache@v4
       with:
         path:  mlir/llvm-project
-        key: Linux-llvm-${{ needs.constants.outputs.llvm_version }}-generic-source
+        key: Linux-llvm-${{ needs.constants.outputs.llvm_version }}-default-source
         enableCrossOsArchive: True
 
     - name: Cache MHLO Source
       id: cache-mhlo-source
       uses: actions/cache@v4
       with:
         path: mlir/mlir-hlo
-        key: Linux-mhlo-${{ needs.constants.outputs.mhlo_version }}-generic-source
+        key: Linux-mhlo-${{ needs.constants.outputs.mhlo_version }}-default-source
         enableCrossOsArchive: True
 
     - name: Cache Enzyme Source
       id: cache-enzyme-source
       uses: actions/cache@v4
       with:
         path:  mlir/Enzyme
-        key: Linux-enzyme-${{ needs.constants.outputs.enzyme_version }}-generic-source
+        key: Linux-enzyme-${{ needs.constants.outputs.enzyme_version }}-default-source
```

Comment:
Is it not macos at the beginning? same for mhlo and llvm above

Reply:
Source code is not OS specific, using the same key allows reusing a single cache for multiple platforms.

Reply:
Then I would remove the Linux- everywhere no?

Reply:
Yup it's a good point, I mainly left linux on my previous pr so I could reuse the one that is already there but might as well change it now :)